### DISCUSSION TITLE
[Steam Deck] Blender Crashes when selecting HIP compute

### DISCUSSION THREAD

**Discussion by Mhowser** [rocm-arch/rocm-arch#955] [Support] [answered]
Hello, 
Not sure if this is the right place to submit this bug report, but since this project is the closest I have ever been to getting the VanGogh GPU working for Blender, I think it fits. 

What I have is **Blender 3.4.1**, downloaded straight off the Official website.  my Steam Deck is running stock **SteamOS 3.4.4** (latest stable) with some tweaks to `pacman` for getting the required packages.  `rocm-hip-sdk` and `rocm-opencl-sdk` + all deps were installed just fine. I was missing the `hsa-amd-aqlprofile-bin` package, fixed that. Now I am stuck here, on this crash when I click on the HIP render device button in Blender settings:
```
$ ./blender
Read prefs: /home/deck/.config/blender/3.4/config/userpref.blend
/run/user/1000/gvfs/ non-existent directory
blender: /build/hsa-rocr/src/ROCR-Runtime-rocm-5.4.1/src/image/addrlib/src/gfx10/gfx10addrlib.cpp:1071: virtual rocr::Addr::ChipFamily rocr::Addr::V2::Gfx10Lib::HwlConvertChipFamily(unsigned int, unsigned int): Assertion `false' failed.
Aborted (core dumped)
```
Here is some relevant information that might help:
[rocminfo.txt](https://github.com/rocm-arch/rocm-arch/files/10365905/rocminfo.txt)
[system-infoCLI.txt](https://github.com/rocm-arch/rocm-arch/files/10365907/system-infoCLI.txt)
[glxinfo.txt](https://github.com/rocm-arch/rocm-arch/files/10365917/glxinfo.txt)
[vulkaninfo.txt](https://github.com/rocm-arch/rocm-arch/files/10365919/vulkaninfo.txt)

Output of [inxi](https://github.com/smxi/inxi) -G:
```
$ ./inxi -G
Graphics:
  Device-1: AMD VanGogh [AMD Custom GPU 0405] driver: amdgpu v: kernel
  Display: x11 server: X.Org v: 21.1.4 with: Xwayland v: 21.1.99 driver: X:
    loaded: amdgpu unloaded: modesetting gpu: amdgpu resolution:
    1: 1920x1080~60Hz 2: 1280x800~60Hz
  OpenGL: renderer: AMD Custom GPU 0405 (vangogh LLVM 14.0.6 DRM 3.45
    5.13.0-valve36-1-neptune) v: 4.6 Mesa 22.2.0 (git-17e5312102)
```
I tried `$ ./blender  --debug-gpu-force-workarounds`, `$ ./blender --env-system-python HSA_OVERRIDE_GFX_VERSION=10.3.0` and `$ HSA_OVERRIDE_GFX_VERSION=10.3.0 ./blender` with no spectacular results, just the same crash.

Please let me know if you require some more information and how to properly get it for you.

**Comment by tpkessler** [comment]
Hey! This is a good place for this issue. Since all involved packages are also in [community-testing] you can also use the official Arch Linux bug tracker. The problem in your case is that mobile GPUs are not officially supported. So some calls (like rocminfo and there-like) may work but more complex code like that generated by blender may fail. See https://github.com/rocm-arch/rocm-arch#package-x-does-not-work

**Comment by Mhowser** [comment]
looking at https://github.com/RadeonOpenCompute/ROCR-Runtime/blob/ac0384d6256b4910509c257a30d1a13d01b8a5ee/src/image/addrlib/src/gfx10/gfx10addrlib.cpp it seems like the VanGogh GPU is supported? (I dont know anything about C++) and the problem might be that the method to get information about the GPU is not sending the correct value. **This is my uneducated guess.**
```LANG=c++
 case FAMILY_VGH:
            if (ASICREV_IS_VANGOGH(chipRevision))
            {
                m_settings.supportRbPlus   = 1;
                m_settings.dccUnsup3DSwDis = 0;
            }
            else
            {
                ADDR_ASSERT(!"Unknown chip revision");
            }
            break;
```

**Comment by tpkessler** [comment, marked as answer]
As long as the architecture is not listed on the official list, there's not much we can do. You can try, but it may fail unexpectedly.

**Comment by syberphunk** [comment]
Fascinating, since the Van Gogh (VN) APU has the Cadence Vision Q6 and Q5 cores.

**Comment by splatchoot** [comment]
Any update on the subject? I can actually select the GPU and click on the render button in the viewport but I end up with this error "HIP hipcc compiler not found. Install HIP toolkit in default location."
